### PR TITLE
[SPARK-50433][DOCS][TESTS] Fix configuring log4j2 guide docs for Spark on YARN and UT

### PR DESCRIPTION
**Pull request by pan3793**
### What changes were proposed in this pull request?

As title.

### Why are the changes needed?

SPARK-37814 (3.3.0) migrated logging system from log4j1 to log4j2, we should updated the docs as well.

### Does this PR introduce _any_ user-facing change?

Yes, docs are updated.

### How was this patch tested?

Review.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

File: connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/test/IntegrationTestUtils.scala [hunk 1]
```diff
@@ -74,7 +74,7 @@ object IntegrationTestUtils {
 
         // Redirect server log into console
         "--conf",
-        s"spark.driver.extraJavaOptions=-Dlog4j.configuration=$log4j2")
+        s"spark.driver.extraJavaOptions=-Dlog4j.configurationFile=$log4j2")
```
Comment (pan3793) [inline connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/test/IntegrationTestUtils.scala:77]: cc @zhenlineo

**Comment by pan3793** [comment]
cc @viirya

File: docs/running-on-yarn.md [hunk 2]
```diff
@@ -126,13 +126,13 @@ settings and a restart of all node managers. Thus, this is not applicable to hos
 
 To use a custom log4j configuration for the application master or executors, here are the options:
```
Comment (viirya) [inline docs/running-on-yarn.md:127]:
```suggestion
To use a custom log4j2 configuration for the application master or executors, here are the options:
```

File: resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnClusterSuite.scala [hunk 3]
```diff
@@ -293,7 +293,7 @@ class YarnClusterSuite extends BaseYarnClusterSuite {
   }
 
   test("running Spark in yarn-cluster mode displays driver log links") {
-    val log4jConf = new File(tempDir, "log4j.properties")
+    val log4jConf = new File(tempDir, "log4j2.properties")
```
Comment (viirya) [inline resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnClusterSuite.scala:296]: Note that this test actually uses `configurationFile` so no matter what the filename of properties file is.

File: connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/test/IntegrationTestUtils.scala (hunk 1 above)
Comment (viirya) [reply connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/test/IntegrationTestUtils.scala:77]: Same question.

**Review by viirya — APPROVED** [review]

File: resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnClusterSuite.scala (hunk 3 above)
Comment (pan3793) [reply resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnClusterSuite.scala:296]: yes, the renaming is just for consistency

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @pan3793 and Liang-Chi.
Merged to master.

**Comment by dongjoon-hyun** [comment]
If you want this in Apache Spark 3.5.x, feel free to open a backport, @pan3793 .

**Comment by pan3793** [comment]
@dongjoon-hyun thanks, I opened https://github.com/apache/spark/pull/49044 for 3.5

**Pull request closed without merging**